### PR TITLE
Swift 4.2 compatibility and removal of conflicting extensions

### PR DESCRIPTION
**Pull request by lessthanyouthink**
Hi there, I’m not sure if this is something you’d want to merge in but here are some changes I just made for a project I’m working on that I just migrated to Xcode 10. 

I removed the extensions that conflict with the Swift 4.2 standard library, and moved over some other functionality to use the standard library stuff. Hope someone else finds this useful 🙂.

### CODE REVIEW

File: SwiftRandom/Randoms.swift [hunk 1]
```diff
@@ -9,46 +9,19 @@ import UIKit
 
 // each type has its own random
 
-public extension Bool {
```
Comment (thellimist) [inline SwiftRandom/Randoms.swift]: what happened to bool?

File: SwiftRandom/Randoms.swift [hunk 2]
```diff
@@ -9,46 +9,19 @@ import UIKit
...
-
     /// SwiftRandom extension
     public static func random(_ lower: Int = 0, _ upper: Int = 100) -> Int {
-        return lower + Int(arc4random_uniform(UInt32(upper - lower + 1)))
+        return Int.random(in: lower...upper)
     }
 }
 
 public extension Int32 {
-    /// SwiftRandom extension
```
Comment (thellimist) [inline SwiftRandom/Randoms.swift]: Why are the range parametered functions deleted?

File: SwiftRandom/Randoms.swift [hunk 3]
```diff
@@ -150,40 +123,11 @@ public extension UIColor {
     }
 }
 
-public extension Array {
-    /// SwiftRandom extension
```
Comment (thellimist) [inline SwiftRandom/Randoms.swift]: + this

File: SwiftRandom/Randoms.swift [hunk 4]
```diff
@@ -150,40 +123,11 @@ public extension UIColor {
...
-
-        let index = Int(arc4random_uniform(UInt32(self.count)))
-        return self[index]
-    }
-}
-
-public extension ArraySlice {
-    /// SwiftRandom extension
-    public func randomItem() -> Element? {
-        guard self.count > 0 else {
```
Comment (thellimist) [inline SwiftRandom/Randoms.swift]: + this

File: SwiftRandom/Randoms.swift (hunk 1 above)
Comment (lessthanyouthink) [reply SwiftRandom/Randoms.swift]: Swift 4.2’s now includes a `random()` function on Bool in the standard library

File: SwiftRandom/Randoms.swift (hunk 2 above)
Comment (lessthanyouthink) [reply SwiftRandom/Randoms.swift]: Swift 4.2 also includes those constructors in the standard library… basically my whole PR was removing what is now offered as a standard language feature 🙂

File: SwiftRandom/Randoms.swift (hunk 3 above)
Comment (lessthanyouthink) [reply SwiftRandom/Randoms.swift]: Replaced by `randomElement()` in 4.2 🙂

**Comment by lessthanyouthink** [comment]
Some additional context: https://github.com/apple/swift-evolution/blob/master/proposals/0202-random-unification.md

**Comment by raysarebest** [comment]
I know this has already been merged, but why aren't the extensions simply wrapped in calls to `#if swift(<= 4.2)`? It seems like this PR unnecessarily drops support for all older versions of Swift

**Pull request merged**